### ISSUE TITLE
Running CellPose Model On Larger Images

### ISSUE THREAD

**Issue by gatordevin** [MouseLand/cellpose#598]
First of all, love this model, and is working great for our series of low signal-to-noise ratio fluorescent images. However, my training folder consists of 300x300 images cropped out of one 2000x4000 image, and these cropped images is what I sued to train the model and do all my annotations. When I have the model that I like and try to run it on the original large image, It doesn't work. I notice that the original cyto model doesn't have such a problem. The cyto model isn't optimized for my dataset but it is good enough to see some cells and the probability map has clearly defined cellular regions. Meanwhile, the custom model I trained has a probability map with no defined cellular areas even though it runs perfectly on my 300x300 images. Any ideas on how to remedy this issue.

**Comment by FrickTobias** [comment]
I would first try this:
1. Have the same aspect ratio in my training data as my target data
2. Use the exact same image size in my training data as my target data

It essentially depends on how the images are augmented in the training process (which I am afraid I have not read myself in the case of CellPose) but it is possible your 2000x4000 images and 300x300 images are resized to the same size before being input into the model. This means your training images are effectively zoomed by a factor of 6.6x13.13 (300/2000x300/4000) compared to your actual data. Again, I haven't read CellPose's image processing code, but in the case of fine-tuning a model it could essentially mean that you are overfitting to your training data, which might mean overfitting to the wrong thing if image format is not the same as in the data you want to deploy your model on.

**Comment by gatordevin** [comment]
Hey. Yeah what I noticed now that I've done some more training is that the model works on the large image but detects about half of the cells it detects on only a 300x300 image. I dove into the code and it appears that when running the model it internally splits it into many 224x224 images it then runs the model on and stitches it together and returns it back. So essentially my training data on 300x300 is very similar to the 224x224 images that are being fed in for testing.

**Comment by FrickTobias** [comment]
Alright, then I guess that shouldn't be an issue. What parameters did you use for training?

**Comment by gatordevin** [comment]
I used the default training parameters offered by the cell pose ui. Soon I'll post a comparison image of how it does when run on a whole image compared to run on one section indepently both with the same threshold and diameter parameters set.

**Comment by fanweiya** [comment]
> I used the default training parameters offered by the cell pose ui. Soon I'll post a comparison image of how it does when run on a whole image compared to run on one section indepently both with the same threshold and diameter parameters set.

Have you got the result?

**Comment by carsen-stringer** [comment]
I am guessing the normalization was the issue then, here is more info: https://github.com/MouseLand/cellpose/issues/608

The solution is to do normalization on all the training images in the folder together instead of one at a time like cellpose does. I will look into implementing this

**Comment by gatordevin** [comment]
I can confirm normalization was the issue. I simply ran the model without normalization of the training data and pre-normalized my split images with my own script. That way no modifications occur to the images before going through the training pipeline and that solved my issues when running the model on the WSI.

**Comment by lisa-melamed** [comment]
@gatordevin We have been having similar issues with running cellpose on larger images and while attempting the fix you suggest here (pre-normalizing images and running the model without normalization), our model returned 0 ROI after taking an inordinate amount of time to run. We suspect its because --no_norm is causing the model to identify too many ROI which crashes the run and returns 0 ROI. Could you share your script that you were able to successfully use?

**Comment by remybonnav** [comment]
> [@gatordevin](https://github.com/gatordevin) We have been having similar issues with running cellpose on larger images and while attempting the fix you suggest here (pre-normalizing images and running the model without normalization), our model returned 0 ROI after taking an inordinate amount of time to run. We suspect its because --no_norm is causing the model to identify too many ROI which crashes the run and returns 0 ROI. Could you share your script that you were able to successfully use?

I also have issue, I can train using the --no_norm flag but it has same effect as if there was normalization when doing inference. And during inference in all cases if I use --no_norm it wil return 0 or only a few ROIs.

Did yo find a way to make it work?
